vue-acme-store/src/components/StoresList.vue

```diff
@@ -0,0 +1,194 @@
+          :class="{ active: index == selectedIndex }"
+          v-for="(store, index) in stores"
+          :key="index"
+          @click="setActiveStore(store, index)"
+        >
+          {{ store.id + ' - ' + store.name }}
+        </li>
+      </ul>
+
+      <b-button class="m-3 btn btn-sm" @click="firstPage" :disabled="currentPage===0">
+        First
+      </b-button>
+
+      <b-button class="m-3 btn btn-sm" @click="previousPage" :disabled="currentPage===0">
+        Previous
+      </b-button>
+
+      <b-button class="m-3 btn btn-sm" @click="nextPage" :disabled="currentPage===totalElements-1">
+        Next
+      </b-button>
+
+      <b-button class="m-3 btn btn-sm" @click="lastPage" :disabled="currentPage===totalElements-1">
+        Last
+      </b-button>
+
+    </div>
+    <div class="col-md-6">
+      <div v-if="currentStore">
+        <h4>Store {{ currentStore.id }}</h4>
+        <div>
+          <label><strong>Name:</strong></label> {{ currentStore.name }}
+          <b-button variant="outline-secondary" class="m-3 btn btn-sm" :href="'/stores/' + currentStore.id + '/' + currentStore.name ">
+            Edit
+          </b-button>
+        </div>
+        <div>
+          <label><strong>Code:</strong></label>
+          {{ currentStore.code }}
+        </div>
+        <div>
+          <label><strong>Store type:</strong></label>
+          {{ currentStore.storeType}}
+        </div>
+        <div>
+          <label><strong>Store season:</strong></label>
+          {{ currentStore.storeSeason}}
+        </div>
+        <div>
+          <label><strong>Opening date:</strong></label>
+          {{ currentStore.openingDate | formatDate }}
+        </div>
+        <div>
+          <label><strong>Description:</strong></label>
+          {{ currentStore.description }}
+        </div>
+        <div>
+          <label><strong>Special field 1:</strong></label>
+          {{ currentStore.specialField1 }}
+        </div>
+        <div>
+          <label><strong>Special field 2:</strong></label>
+          {{ currentStore.specialField2 }}
+        </div>
+      </div>
+      <div v-else>
+        <br />
+        <p>Please click on a Store...</p>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Vue } from "vue-property-decorator";
+import StoreDataService from "../services/StoreDataService";
+
+@Component
+export default class StoresList extends Vue {
+  private name: string = "";
+
+  private currentStore: any = null;
+  private selectedIndex: number = -1;
+  private selectedPage: number = 0;
+
+  private currentPage: number = 0;
+  private stores: any[] = [];
+  private totalElements: number = 0;
+
+
+  retrieveStores() {
+    StoreDataService.getAll(this.currentPage)
+      .then((response) => {
+        this.stores = response.data.data.map(this.mapperStore);
+        this.totalElements = response.data.totalElements;
+        console.log(response.data.data);
+      })
+      .catch((e) => {
```

Comment:
besides having error handling, is there any other thing that you think it might be missing on these requests?